terraform/main.tf

```
@@ -301,6 +332,213 @@ resource "aws_iam_role_policy" "api_task-publish_to_arpa_audit_report_queue" {
+  launch_type             = "FARGATE"
+  enable_ecs_managed_tags = true
+  enable_execute_command  = false
+
+  task_override = jsonencode({
+    containerOverrides = [
+      {
+        name = "api"
+        command = [
+          "node",
+          "-e",
+          "require('./src/lib/grants-digest').run().then(() => { process.exit(0); }).catch((err) => { console.log(err); process.exit(1); });"
+        ]
+        environment = [
+          {
+            name  = "ENABLE_DIGEST_EMAIL_CRITERIA_CRON",
+            value = "true"
+          },
+        ]
+      },
+    ]
+  })
+
+  network_configuration = {
+    assign_public_ip = false
+    security_groups  = [module.digest_email_kickoff_cron_security_group.id]
+    subnets          = var.subnet_ids
+  }
+}
+
+data "aws_iam_policy_document" "publish_to_digest_email_get_grants" {
+  statement {
+    sid       = "AllowPublishToQueue"
+    actions   = ["sqs:SendMessage"]
+    resources = [module.digest_email_get_grants.sqs_queue_arn]
+  }
+}
+
+resource "aws_iam_role_policy" "digest_email_kickoff_cron-publish_to_digest_email_get_grants_queue" {
+  name_prefix = "send-digest-email-get-grants-requests"
+  role        = module.digest_email_kickoff_cron.ecs_task_role_name
+  policy      = data.aws_iam_policy_document.publish_to_digest_email_send_queue.json
+}
+
+module "digest_email_get_grants" {
+  source                   = "./modules/sqs_consumer_task"
+  namespace                = "${var.namespace}-digest_get_grants"
+  permissions_boundary_arn = local.permissions_boundary_arn
+
+  # Networking
+  subnet_ids         = local.private_subnet_ids
+  security_group_ids = [module.digest_email_get_grants_for_criteria_security_group.id]
+
+  # Task configuration
+  ecs_cluster_name      = join("", aws_ecs_cluster.default.*.name)
+  docker_tag            = var.api_container_image_tag
+  unified_service_tags  = local.unified_service_tags
+  stop_timeout_seconds  = 120
+  consumer_task_command = ["node", "./src/lib/digest/getGrantsAndUsers.js"]
+  consumer_container_environment = {
+    API_DOMAIN          = local.api_domain_name
+    DATA_DIR            = "/var/data"
+    NODE_OPTIONS        = "--max_old_space_size=3584" # Reserve 512 MB for other task resources
+    NOTIFICATIONS_EMAIL = "grants-notifications@${var.website_domain_name}"
+    WEBSITE_DOMAIN      = "https://${var.website_domain_name}"
+  }
+  consumer_task_efs_volume_mounts = [{
+    name            = "data"
+    container_path  = "/var/data"
+    read_only       = false
+    file_system_id  = module.api.efs_data_volume_id
+    access_point_id = module.api.efs_data_volume_access_point_id
+  }]
+  # Task resource configuration
+  # TODO: Tune these values after observing usage in different environments.
+  #       See also: --max_old_space_size in NODE_OPTIONS env var.
+  consumer_task_size = {
+    cpu    = 1024 # 1 vCPU
+    memory = 4096 # 4 GB
+  }
+
+  # Messaging
+  autoscaling_message_thresholds = [1, 3, 5, 10, 20, 50]
```

Unlike the `arpa_audit_report` task, where simultaneous requests for reports are rare and we need to generate them in an hour or less, we can probably assume this task's source queue will generally have a high starting message count (whenever it isn't empty) and doesn't need to work through the queue as fast as possible. These thresholds set the number of tasks based on the number of messages in the source queue, so low values like `1, 3, 5` aren't worth setting in this particular use-case. Individual jobs (per-message) will also probably finish pretty swiftly, so I would tune this more along the values we expect to see (i.e. the number of unique search criteria).

Something like this is probably a decent starting point:
```suggestion
  autoscaling_message_thresholds = [200, 500, 1000, 2000, 5000, 10000]
```

NB: That :point_up: is 1 task at 0-200 messages, working up to 7 tasks at anything > 10k messages.